mutis/signal.py

```diff
@@ -178,21 +179,20 @@ def PSD_check_gen(self, fgen=None, ax=None):
         # TODO: make a generic check_gen method
 
         if fgen is None:
-            fgen = self.method
+            fgen = self.fgen
         if ax is None:
             ax = plt.gca()
-        if fgen == "lc_gen_psd":
-            s2 = lc_gen_psd(self.times, self.signs)
-        elif fgen == "lc_gen_psd_nft":
-            s2 = lc_gen_psd_nft(self.times, self.signs)
+
+        if fgen == "lc_gen_psd_nft":
+            s2 = lc_gen_psd_nft(self.times, self.values)
         elif fgen == "lc_gen_psd_lombscargle":
-            s2 = lc_gen_psd_lombscargle(self.times, self.signs)
+            s2 = lc_gen_psd_lombscargle(self.times, self.values)
         elif fgen == "lc_gen_psd_fft":
-            s2 = lc_gen_psd_fft(self.signs)
+            s2 = lc_gen_psd_fft(self.values)
         else:
```

Comment:
Unnecessary "else" after "raise".
Please do it in this way:

```Python
elif self.fgen == "lc_gen_ou":
        if self.theta is None or self.mu is None or self.sigma is None:
                raise Exception("You need to set the parameters for the signal")
        self.synth[n] = lc_gen_ou(self.theta, self.mu, self.sigma, self.times)

```

Reply:
Yes I think we could rename this `N` to something more explicit.  This `Signal().N` is the number of generated synthethic light curves... so maybe `.N_synth`?

As for your comment, I do not see an `else` after a `raise` in line 191, it is a `raise` after an `else`... If the fgen is not any of them, then we raise an exception. I don't understand how you propose to change it, can you write it?

Reply:
oups, sorry I placed the comment in the wrong line, I have hopefully copied to the right one now. as for the name I think `samples` is better than `N_synth`

Reply:
Done